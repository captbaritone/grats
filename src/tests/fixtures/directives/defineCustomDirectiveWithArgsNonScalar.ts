/** @gqlInput */
type SomeInput = {
  someField: string;
};

/**
 * This is my custom directive.
 * @gqlDirective
 * @on FIELD_DEFINITION
 */
export function customDirective(args: { someArg: SomeInput }) {}
