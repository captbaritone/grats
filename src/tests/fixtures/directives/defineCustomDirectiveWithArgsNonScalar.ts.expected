-----------------
INPUT
----------------- 
/** @gqlInput */
type SomeInput = {
  someField: string;
};

/**
 * This is my custom directive.
 * @gqlDirective
 * @on FIELD_DEFINITION
 */
export function customDirective(args: { someArg: SomeInput }) {}

-----------------
OUTPUT
-----------------
-- SDL --
"""This is my custom directive."""
directive @customDirective(someArg: SomeInput!) on FIELD_DEFINITION

input SomeInput {
  someField: String!
}
-- TypeScript --
import { GraphQLSchema, GraphQLDirective, DirectiveLocation, GraphQLNonNull, GraphQLInputObjectType, GraphQLString } from "graphql";
export function getSchema(): GraphQLSchema {
    const SomeInputType: GraphQLInputObjectType = new GraphQLInputObjectType({
        name: "SomeInput",
        fields() {
            return {
                someField: {
                    name: "someField",
                    type: new GraphQLNonNull(GraphQLString)
                }
            };
        }
    });
    return new GraphQLSchema({
        directives: [new GraphQLDirective({
                name: "customDirective",
                locations: [DirectiveLocation.FIELD_DEFINITION],
                description: "This is my custom directive.",
                args: {
                    someArg: {
                        type: new GraphQLNonNull(SomeInputType)
                    }
                }
            })],
        types: [SomeInputType]
    });
}
